test/test_torch.py

```diff
@@ -10898,6 +10898,11 @@ def test_empty_strided(self, device):
                 self.assertEqual(empty_strided.shape, as_strided.shape)
                 self.assertEqual(empty_strided.stride(), as_strided.stride())
 
+    def test_strided_mismatched_stride_shape(self, device):
+        for shape, strides in [((1, ), ()), ((1, 2), (1, ))]:
+            with self.assertRaisesRegex(RuntimeError, "mismatch in length of strides and shape"):
+                torch.tensor(0.42, device=device).as_strided(shape, strides)
+
```

Comment:
@ailzhang 

`RuntimeError` is not thrown with XLA.
Should we have something similar to what `test_vander` used to have,
```python
if self.device_type != 'XLA':
    # Test Code
```